test/groovy/com/sap/piper/cm/ChangeManagementTest.groovy

```diff
@@ -220,6 +220,42 @@ public void testGetCommandLineWithCMClientOpts() {
             'openSesame')
     }
 
+    @Test
+    public void testReleaseTransportRequestSucceeds() {
+
+        // the regex provided below is an implicit check that the command line is fine.
+        script.setReturnValue(JenkinsShellCallRule.Type.REGEX, 'release-transport.*-cID 001.*-tID 002', 0)
+
+        new ChangeManagement(nullScript).releaseTransportRequest('001',
+            '002',
+            'https://example.org',
+            'me',
+            'openSesame')
+
+        // no assert required here, since the regex registered above to the script rule is an implicit check for
+        // the command line.
+    }
+
+    @Test
+    public void testReleaseTransportRequestFails() {
+
+        thrown.expect(ChangeManagementException)
+        thrown.expectMessage("Cannot release Transport Request '002'. Return code from cmclient: 1.")
+
+        // the regex provided below is an implicit check that the command line is fine.
+        script.setReturnValue(JenkinsShellCallRule.Type.REGEX, 'release-transport.*-cID 001.*-tID 002', 1)
+
+        new ChangeManagement(nullScript).releaseTransportRequest('001',
+            '002',
+            'https://example.org',
+            'me',
+            'openSesame')
+
+        // no assert required here, since the regex registered above to the script rule is an implicit check for
```

Comment:
the comment contradicts to the exception expected

Reply:
Why? In case the command line does not match there is no integer '1' returned, another Exception (currently NPE) is thrown. Hence the test fails since we get an exception of the wrong type.

Maybe I did not get your point. In this case: can you explain more details?

Reply:
i'd say: no assert required, since an exception is expected.

Reply:
OK, if an exception is expected there is - normally - no assert. We assert the exception, so to say ...

The point is: the exception which we expect is only triggered in case the regex registered to the shell rule is seen by the shell rule - this is what I mean with implicit check by the shell rule. If - for any reason, e.g. shuffled parameters when calling the cm util - the shell call looks different, the return code != 0 is not returned by the shell rule, in turn the ChangeManagementException is not triggered, hence the ChangeManagementException cannot be catched and as a result the AbortExcpetion is not triggered ...